core/src/main/scala/org/apache/spark/TaskEndReason.scala

```diff
@@ -208,6 +208,22 @@ case class ExecutorLostFailure(execId: String) extends TaskFailedReason {
 
 /**
  * :: DeveloperApi ::
+ * The task failed because the executor that it was running on was prematurely terminated. The
+ * executor is forcibly exited but the exit should be considered as part of normal cluster
+ * behavior.
+ */
+@DeveloperApi
+case class ExecutorNormalExit(
```

Comment:
I'd give the same feedback here, but then `ExecutorLostFailure` is a developer API... still I think that a single reason (with a boolean saying whether to treat it as an error) would be simpler.